src/main/groovy/life/qbic/portal/sampletracking/web/presenters/ControlElementsPresenter.groovy

```diff
@@ -22,11 +22,11 @@ class ControlElementsPresenter implements QueryAvailableLocationsOutput {
 
     @Override
     def updateAvailableLocations(List<Location> locations) {
-        return null
+        this.viewModel.availableLocations.addAll(locations ?: [])
     }
 
     @Override
     def invokeOnError(String msg) {
-        return null
+        this.viewModel.failureNotifications << msg
```

Comment:
I like the `.add` method better since it is more readable and common in Java programs.
```suggestion
        this.viewModel.failureNotifications.add(msg)
```